### DISCUSSION TITLE
How to search with image and text embeddings?

### DISCUSSION THREAD

**Discussion by Talhaz** [rom1504/clip-retrieval#51] [Q&A]
`import faiss
import numpy as np
ind = faiss.read_index("/content/index_folder_jumpsuit/image.index")
D, I = ind.search(text_embeddings, 5)`
in this search function only one embeddings u can pass how to pass image embeddings with text embeddings  ?

**Comment by rom1504** [comment]
Interesting question!
That's something I want to support in the near future.
One way to do it is to concatenate your text and image embeddings and build and index out of it and then query with concatenation.

If you only have an image index and want to query with both, you might try to query with the average of your image and text embeddings

**Reply by ytzeng1** [reply]
Any progress on this? It would be a useful feature for my application. I can look into implementing this as well if you prefer.
